Sources/MessagingPush/MessagingPush.swift

```diff
@@ -36,16 +36,31 @@ public class MessagingPush: ModuleTopLevelObject<MessagingPushInstance>, Messagi
     public static func initialize(
         configure configureHandler: ((inout MessagingPushConfigOptions) -> Void)? = nil
     ) -> MessagingPushInstance {
-        var configOptions = moduleConfig
-
         if let configureHandler = configureHandler {
-            configureHandler(&configOptions)
+            // pass current config reference to update it without needing to recreate
+            configureHandler(&moduleConfig)
         }
 
         shared.initializeModule()
         return shared
     }
 
+    /// MessagingPush initializer for Notification Service Extension
```

Comment:
From my understanding, customers are asked to call `CustomerIO.initialize()` to provide the `writeKey` in the host app. Then, the `writeKey` can be referenced by other modules in the code. 

This change looks like we are introducing `writeKey` into the push module. This `writeKey` would only be used by the NSE for delivered push metrics. 

Assuming this is correct, could we avoid introducing `writeKey` to the push module? Having to manage multiple `writeKeys` in the SDK, depending on the context, would not be ideal. I would consider this change as introducing technical debt and I would prefer we avoid it if possible.